package/MDAnalysis/core/topologyattrs.py

```diff
@@ -1373,10 +1479,10 @@ def set_atoms(self, ag):
     def get_atoms(self, ag):
         try:
             unique_bonds = set(itertools.chain(
-                *[self._bondDict[a] for a in ag._ix]))
+                *[self._bondDict[a] for a in ag.ix]))
         except TypeError:
             # maybe we got passed an Atom
-            unique_bonds = self._bondDict[ag._ix]
+            unique_bonds = self._bondDict[ag.ix]
         bond_idx, types, guessed, order = np.hsplit(
             np.array(sorted(unique_bonds)), 4)
         bond_idx = np.array(bond_idx.ravel().tolist(), dtype=np.int32)
```

Comment:
(Not part of the PR but is `.tolist()` needed? Couldn't you do `np.asarray(bond_idx.ravel(), dtype=np.int32)` ?)